File: 1925.count-square-sum-triples/solution.go
func countTriples(n int) int {
    res := 0
    for a := 1; a <= n; a++ {
        for b := 1; b <= n; b++ {
            target := a * a + b * b
            c := sort.Search(n + 1, func(c int) bool {
                return c * c >= target
            })
            if c <= n && c * c == target {
                res++
            }
        }
    }
    return res
}

File: 1925.count-square-sum-triples/solution.kt
class Solution {
    fun countTriples(n: Int): Int {
        var res = 0
        for (a in 1 .. n) {
            for (b in 1 .. n) {
                for (c in 1 .. n) {
                    if (a * a + b * b == c * c) {
                        res++
                    }
                }
            }
        }
        return res
    }
}